### PR TITLE
Readability pull request

### PR DESCRIPTION
**Pull request by jbambas**
Added Readability plugin that allows reading mode required in issue #1675.

### CODE REVIEW

**Comment by nowrep** [comment]
Please remove the translation files (*.ts), they are for TestPlugin.

File: plugins/Readability/data/Call.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+function readabilityHead(article){
+	var head = document.createElement('head');
+	document.documentElement.appendChild(head);
+
+	var title = document.createElement('title');
+	title.innerHTML = article.title;
+	head.appendChild(title);
+
+    var css = document.createElement('style');
```
Comment (nowrep) [inline plugins/Readability/data/Call.js]: Please fix indentation. Somewhere you use tabs and somewhere spaces, I would prefer to use only spaces.

File: plugins/Readability/data/Call.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	if(element && !!article && !!article.content){
+		element.removeChild(document.body);
+		element.removeChild(document.head);
+
+		readabilityHead(article);
+		readablilityBody(article);
+
+		readablilityToolbar();
+	} else {
+		alert("No content to transform.");
```
Comment (nowrep) [inline plugins/Readability/data/Call.js]: This needs to be translated.

File: plugins/Readability/data/style.css [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+}
+
+.dropdown.open,
+.button:active,
+#font-size-buttons > button:active,
+#font-size-buttons > button.selected {
+    background-color: #dadada;
+}
+
+/* Only used on Android */
```
Comment (nowrep) [inline plugins/Readability/data/style.css]: If this is only used on Android, then please remove it.

File: plugins/Readability/data/Toolbar.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
+function readablilityToolbar(){
```
Comment (nowrep) [inline plugins/Readability/data/Toolbar.js]: This file is missing copyright.

File: plugins/Readability/data/RM-Type-Controls-Arrow.svg [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nowrep) [inline plugins/Readability/data/RM-Type-Controls-Arrow.svg]: If this file is not used, please remove.

File: plugins/Readability/data/Call.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
+%1
```
Comment (nowrep) [inline plugins/Readability/data/Call.js]: Also missing copyright.

File: plugins/Readability/readability.cpp [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+#include "webview.h"
+#include "webpage.h"
+
+#include <QMenu>
+#include <QTranslator>
+#include <QPushButton>
+
+ReadabilityPlugin::ReadabilityPlugin()
+    : QObject(),
+    m_view(0)
```
Comment (nowrep) [inline plugins/Readability/readability.cpp]: `, m_view(0)`

File: plugins/Readability/readability.cpp [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+void ReadabilityPlugin::init(InitState state, const QString &settingsPath)
+{
+    m_settingsPath = settingsPath;
+    Q_UNUSED(state)
+}
+
+void ReadabilityPlugin::unload()
+{}
```
Comment (nowrep) [inline plugins/Readability/readability.cpp]:
```
{
}
```

File: plugins/Readability/readability.cpp [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    return translator;
+}
+
+void ReadabilityPlugin::showSettings(QWidget* parent)
+{
+}
+
+void ReadabilityPlugin::populateWebViewMenu(QMenu* menu, WebView* view, const WebHitTestResult &r)
+{
+    m_view=view;
```
Comment (nowrep) [inline plugins/Readability/readability.cpp]: `m_view = view`

File: plugins/Readability/readability.cpp [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+    QString m_close = QzTools::pixmapToDataUrl(QPixmap(":/readability/data/RM-close.png")).toString();
+    QString m_close_hover = QzTools::pixmapToDataUrl(QPixmap(":/readability/data/RM-close-hover.png")).toString();
+    QString m_delete = QzTools::pixmapToDataUrl(QPixmap(":/readability/data/RM-Delete-24x24.png")).toString();
+    QString m_plus = QzTools::pixmapToDataUrl(QPixmap(":/readability/data/RM-Plus-24x24.png")).toString();
+    QString m_minus = QzTools::pixmapToDataUrl(QPixmap(":/readability/data/RM-Minus-24x24.png")).toString();
+    QString m_arrow = QzTools::readAllFileContents(":/readability/data/RM-Type-Controls-Arrow.png");
+    QString m_controls = QzTools::pixmapToDataUrl(QPixmap(":/readability/data/RM-Type-Controls-24x24.png")).toString();
+    
+    QString css = QzTools::readAllFileContents(":/readability/data/style.css").arg(m_close, m_close_hover, m_delete, m_plus, m_minus, m_arrow, m_controls);
+    QStringList list = css.split(QRegExp("[\r\n]"), QString::SkipEmptyParts);
```
Comment (nowrep) [inline plugins/Readability/readability.cpp]: This is very inefficient, `.remove(QLatin1Char('\n')` should be enough.

**Comment by cranes-bill** [comment]
Just to say - i'm receiving the following warning from the compiler:
```
readability.cpp:73:6: attention : unused parameter ‘parent’ [-Wunused-parameter]
 void ReadabilityPlugin::showSettings(QWidget* parent)
      ^
```

**Pull request merged**